doc/architecture.md

```diff
@@ -0,0 +1,256 @@
+## Architecture
```

Comment:
Similarly to advanced_topics.md, this document should be linked from the top README.md.

Reply:
Readme updated. Fix c45ba05